tests/endpoints/test_cargo_movements_real.py

```diff
@@ -47,6 +47,71 @@ def test_search_single_filter_id(self):
 
         assert len(df) == 2
 
+    def test_exlusion_filter(self):
+        crude = [
+            p.id
+            for p in Products().search("Crude/Condensates").to_list()
+            if p.layer == ["group"]
+        ]
+        arab_medium = [
+            p.id
+            for p in Products().search("Arab Medium").to_list()
+            if p.layer == ["grade"]
+        ]
+        meg = [
+            g.id
+            for g in Geographies().search("MEG/AG").to_list()
+            if "trading_region" in g.layer
+        ]
+        iraq = [
+            g.id
+            for g in Geographies().search("Iraq").to_list()
+            if "country" in g.layer
+        ]
+
+        cols = [
+            "cargo_movement_id",
+            "vessels.0.name",
+            "events.cargo_port_load_event.0.location.port.id",
+            "events.cargo_port_load_event.0.location.port.label",
+            "events.cargo_port_load_event.0.location.country.id",
+            "events.cargo_port_load_event.0.location.country.label",
+            "events.cargo_port_load_event.0.start_timestamp",
+            "events.cargo_port_load_event.0.end_timestamp",
+            "product.grade.id",
+            "product.grade.label",
+            "product.group_product.label",
+        ]
+
+        df = (
+            CargoMovements()
+            .search(
+                filter_activity="loading_end",
+                filter_products=[
+                    "54af755a090118dcf9b0724c9a4e9f14745c26165385ffa7f1445bc768f06f11"
+                ],
+                filter_origins=meg,
+                exclude={
+                    "filter_origins": iraq[0],
+                    "filter_products": arab_medium,
+                },
```

Comment:
I think here we could make this even simpler and present `exclude_product = arab_medium` as a top level argument to the user. Creating a dictionary of exclusions is an API implementation detail, and something we can take care of internally in the SDK.

Reply:
That's a really good point, you are right. Will try this.

Reply:
```
def wrap_args_to_dict(foo, keyword='exclude') -> Dict:
    """
    Paremeters
    ----------
        foo: function
        keyword: keeps only arguments that contain this keyword
    """
    # Store function args and deafault values
    obj = inspect.getfullargspec(foo)
    din = dict(zip(obj.args, obj.defaults))
    
    # Filter dict based on keyword
    d = dict()
    for k,v in din.items():
        if keyword in k:
            d[k] = convert_to_list(v)
            
    return d
    ```

Reply:
@KitBurgess Before I make any commits, just for some feedback, does the following implementation look fine?
```
def wrap_args_to_dict(foo, keyword='exclude') -> Dict:
    """
    Paremeters
    ----------
        foo: function
        keyword: keeps only arguments that contain this keyword
    """
    # Store function args and deafault values
    obj = inspect.getfullargspec(foo)
    din = dict(zip(obj.args, obj.defaults))
    
    # Filter dict based on keyword
    d = dict()
    for k,v in din.items():
        if keyword in k:
            d[k] = convert_to_list(v)
            
    return d
```

So if we have a function `foo` like this:

```
def foo(exclude_origins='3', exclude_charterers='4', exclude_owners=None, c=5):
    obj = inspect.getfullargspec(test)
    d = dict(zip(obj.args, obj.defaults))
    return d
```

When you call `wrap_args_to_dict(foo,'exclude')` you would get:

```
{'exclude_origins': ['3'], 'exclude_charterers': ['4'], 'exclude_owners': []}
```

So you would then go to `CargoMovements` class and in the `params` use:

```
"exclude": wrap_args_to_dict(search)
```

Reply:
Sorry - I'm not sure I follow this, it looks more complicated though than explicitly creating the `exclude` dictionary, and then filtering this / wrapping the values in a dict.

I think we should commit a working simple solution first, then look to reduce any duplicated code afterwards.

Reply:
The `wrap_args_to_dict` function does the following: It finds all the arguments and default values of a given input function, it then creates a dict where the dict keys are the function arguments and the dict values the default arg values and finally filters the dict by keeping all the pairs where the keys include the specified keyword. 

In our case when we call this function inside the `CargoMovements` class like wrap_args_to_dict(search), it will create the corresponding dict with all arguments that include 'exclude'.

Of course, the other way as you mention is to just define an exclusion dictionary before define the params dict. We would just need to do this in every class, and if for some reasons an endpoint has different arguments than another (e.g. `VesselMovements` has the `filter_vessel_class` filter already deployed, whereas `CargoMovements` not) it would be more difficult I think to maintain.